src/providers/cambridge/functions.py

```diff
@@ -0,0 +1,53 @@
+import os
+import requests
+from bs4 import BeautifulSoup
+import random
+
+def get_page_content(url):
+    response = requests.get(url)
+    if response.status_code == 200:
+        return response.text
+    else:
+        print(f"Error: Failed to fetch the webpage ({response.status_code})")
+        return None
```

Comment:
```suggestion
def get_page_content(url):
    response = requests.get(url)
    try:
        r = requests.get('http://www.google.com/nothere')
        r.raise_for_status()
        return r.text
    except requests.exceptions.HTTPError as err:
        print(f"Error: Failed to fetch the webpage ({response.status_code})")
        return None
```

Also, since it's a cli it's better to use click's ` echo` function

Reply:
With this code, I get "Error: Failed to fetch the webpage (200)" and the rest of the code does not run.